app/common/CustomThemeProvider.tsx

```diff
@@ -0,0 +1,59 @@
+"use client";
+import {
+  SDSLightAppTheme,
+  SDSDarkAppTheme,
+  makeThemeOptions,
+} from "@czi-sds/components";
+import {
+  createTheme,
+  ThemeProvider as MuiThemeProvider,
+  Theme,
+} from "@mui/material";
+import { ReactNode, useState, useEffect } from "react";
+import { customTheme, primaryColors } from "./theme";
+import { ThemeMode, ThemeContext } from "../context/themeContext.context";
+
+// CustomThemeProvider component to wrap your app
+export const CustomThemeProvider = ({ children }: { children: ReactNode }) => {
+  const [themeMode, setThemeMode] = useState<ThemeMode>("light");
+  const [theme, setTheme] = useState<Theme>();
+
+  useEffect(() => {
+    const mediaQuery = window.matchMedia("(prefers-color-scheme: dark)");
+
+    // Set initial theme based on system preference
+    setThemeMode(mediaQuery.matches ? "dark" : "light");
+
+    const handleThemeChange = (e: MediaQueryListEvent) => {
+      setThemeMode(e.matches ? "dark" : "light");
+    };
+
+    // Listen for theme changes
+    mediaQuery.addEventListener("change", handleThemeChange);
+
+    // Clean up event listener on unmount
+    return () => {
+      mediaQuery.removeEventListener("change", handleThemeChange);
+    };
+  }, []);
+
+  useEffect(() => {
```

Comment:
discussion:

We might be able to save a useEffect hook by moving the theme update directly in `handleThemeChange` above

What do you think 😄 ?

Reply:
Sounds good to me.  Moving...

Reply:
🤔 This actually made it unnecessary to have themeMode stored in state at all, so I removed it and also removed the context.  I think this is more clear, but lmk if you disagree.

Reply:
Totally! Love it thank you 🤩 !!